routes/users.js

```diff
@@ -400,17 +410,48 @@ router.delete("/:userId", async (req, res) => {
 
 // PATCH
 router.patch("/:userId", async (req, res) => {
+  const body = req.body;
+
   try {
-    const updatedUser = await User.updateOne(
+    if (body.postalCode) {
+      const location = await LocationUtil.getLocationByPostalCode(
+        body.postalCode
+      );
+      body.location = location;
+
+      const postingsToUpdate = await Posting.updateMany(
+        {
+          ownerId: req.params.userId,
+          active: true,
+        },
+        {
+          location: location,
+        }
+      );
+    }
+
+    const updatedUser = await User.findOneAndUpdate(
       { _id: req.params.userId },
-      { $set: req.body }
+      { $set: body }, 
+      {new: true}
     );
-    res.status(200).json(updatedUser);
+    const token = jwt.sign(updatedUser.toJSON(), process.env.TOKEN_SECRET);
```

Comment:
updating the token when the user updates to fix the refresh error that returns stale user